src/main/java/com/github/mc1arke/sonarqube/plugin/ce/pullrequest/bitbucket/BitbucketPullRequestDecorator.java

```diff
@@ -116,31 +132,24 @@ private CreateReportRequest toReport(AnalysisDetails analysisDetails) {
         reportData.add(securityReport(rules.get(RuleType.VULNERABILITY), rules.get(RuleType.SECURITY_HOTSPOT)));
         reportData.add(new ReportData("Duplication", new DataValue.Percentage(newDuplication(analysisDetails))));
         reportData.add(maintainabilityReport(rules.get(RuleType.CODE_SMELL)));
-        reportData.add(new ReportData("Analysis details", new DataValue.Link("Go to SonarQube", analysisDetails.getDashboardUrl())));
-
-        return new CreateReportRequest(reportData,
-                reportDescription(analysisDetails),
-                "SonarQube",
-                "SonarQube",
-                analysisDetails.getAnalysisDate().toInstant(),
-                analysisDetails.getDashboardUrl(),
-                format("%s/common/icon.png", analysisDetails.getBaseImageUrl()),
-                asInsightStatus(analysisDetails.getQualityGateStatus()));
+        reportData.add(new ReportData("Analysis details", client.get().createLinkDataValue(analysisDetails.getDashboardUrl())));
+
+        return reportData;
     }
 
     private void updateAnnotations(String project, String repo, AnalysisDetails analysisDetails) throws IOException {
         final AtomicInteger chunkCounter = new AtomicInteger(0);
 
-        client.deleteAnnotations(project, repo, analysisDetails.getCommitSha());
+        client.get().deleteAnnotations(project, repo, analysisDetails.getCommitSha());
 
-        Map<Object, Set<Annotation>> annotationChunks = analysisDetails.getPostAnalysisIssueVisitor().getIssues().stream()
+        Map<Object, Set<CodeInsightsAnnotation>> annotationChunks = analysisDetails.getPostAnalysisIssueVisitor().getIssues().stream()
                 .filter(i -> i.getComponent().getReportAttributes().getScmPath().isPresent())
                 .filter(i -> i.getComponent().getType() == Component.Type.FILE)
                 .filter(i -> OPEN_ISSUE_STATUSES.contains(i.getIssue().status()))
                 .sorted(Comparator.comparing(a -> Severity.ALL.indexOf(a.getIssue().severity())))
                 .map(componentIssue -> {
                     String path = componentIssue.getComponent().getReportAttributes().getScmPath().get();
-                    return new Annotation(componentIssue.getIssue().key(),
+                    return client.get().createCodeInsightsAnnotation(componentIssue.getIssue().key(),
```

Comment:
Client simplification:
```suggestion
                    return client.createCodeInsightsAnnotation(componentIssue.getIssue().key(),
```